lib/Url/UrlManager.php

```diff
@@ -281,10 +281,11 @@ public static function resolveUrl(Url $url)
             return null;
         }
 
+        // Weiterleitung auf URL mit Suffix, wenn Suffix fehlt
         $rewriterSuffix = Url::getRewriter()->getSuffix();
         if (\rex::isFrontend() && $rewriterSuffix && substr($url->getPath(), -strlen($rewriterSuffix)) !== $rewriterSuffix) {
             header('HTTP/1.1 301 Moved Permanently');
-            header('Location: '.$url->getPath().$rewriterSuffix.$url->getQuery());
+            header('Location: '.$url->getPath().$rewriterSuffix.($url->getQuery() != "" ? "?".$url->getQuery() : ""));
```

Comment:
Ich würde gern das manuelle schreiben von `?` vermeiden wollen. Die Uri class kann eine korrekte Url mit Parametern liefern.

Reply:
@TobiasKrais könntest du das Feedback von @tbaddade noch umsetzen? Stolpere gerade wieder über das Problem mit den fehlenden Umleitungen.

Reply:
Oh, die Antwort ist total untergegangen. Mein Papa ist  Ende Januar verstorben.
Hab mir das gerade angeschaut. Ich brauche Hilfe, wie ich das mit der URI Klasse lösen kann. Hat mir jemand vielleicht  Beispielcode?

Reply:
Ich glaub ich habs. Wenn ich die Zeile so ändere:
`$uri = new \Riimu\Kit\UrlParser\Uri($url, \Riimu\Kit\UrlParser\UriParser::MODE_UTF8);
header('Location: '. $uri->withQuery($uri->getQuery()));`
funktioniert alles gut außer eine einzige Seite: sitemap.xml. Beim Aufruf erhalte ich einen ERR_TOO_MANY_REDIRECTS.
Lasse ich aber die Zeile so wie vorgeschlagen, dann leitet er mir die sitemap.xml auf die sitemap.xml/ weiter?! Habt ihr eine Idee?

Reply:
Korrektur: es funktioniert nicht alles. Bei `$uri->withQuery($uri->getQuery())` wird der trailing slash nicht angehängt.

Reply:
Kannst du mir hier paar Beispiel Urls notieren. Scheinbar funktionieren einige, andere wieder nicht.

Reply:
Soll ich dir URLs mit der `$uri->withQuery($uri->getQuery())` Änderung schicken oder so wie es im Code des Commits drin ist?

Reply:
```
$uri = new \Riimu\Kit\UrlParser\Uri($url, \Riimu\Kit\UrlParser\UriParser::MODE_UTF8);
header('Location: '. $uri->withQuery($uri->getQuery()));
```
Das kann ja auch nicht funktionieren, denn die URL die ich an die Uri Klasse übergebe hat keinen trailing slash drin. Und die Uri Klasse weiß auch nicht, dass sie ihn einfügen sollte. Um es zu lösen, müsste man die URL bei der Erstellung des Uri Objektes anpassen. Dann sind wir aber wieder bei dem Code im Commit.

Reply:
Die einzige Seite, die nicht funktioniert hatte war die sitemap.xml. Die robots.txt ha funktioniert und es wurde nicht auf robots.txt/ weitergeleitet. Das Problem mit der sitemap.xml habe ich mit dem neuesten Commit gelöst.

Reply:
@TobiasKrais Könntest du das mal ausprobieren

```suggestion
            $url->handleRewriterSuffix();
            header('Location: '.$url->__toString());
```

Reply:
Das ist wunderbar! Genau so funktioniert es!